src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -67,108 +54,132 @@ func Setup(version string, buildDate string) *rotator.SizeRotator {
 			}
 		}
 	}
-	file := rotator.NewSizeRotator(filepath.Join(logdir, "lantern.log"))
+	logFile = rotator.NewSizeRotator(filepath.Join(logdir, "lantern.log"))
 	// Set log files to 1 MB
-	file.RotationSize = 1 * 1024 * 1024
+	logFile.RotationSize = 1 * 1024 * 1024
 	// Keep up to 20 log files
-	file.MaxRotation = 20
+	logFile.MaxRotation = 20
 
 	// Loggly has its own timestamp so don't bother adding it in message,
 	// moreover, golog always write each line in whole, so we need not to care about line breaks.
-	errorOut := timestamped(NonStopWriter(os.Stderr, file))
+	errorOut = timestamped(io.MultiWriter(os.Stderr, logFile))
+	debugOut = timestamped(io.MultiWriter(os.Stdout, logFile))
+	golog.SetOutputs(errorOut, debugOut)
+}
 
+func Configure(cfg *config.Config, version string, buildDate string) {
 	if logglyToken == "" {
 		log.Debugf("No logglyToken, not sending error logs to Loggly")
-	} else {
-		log.Debugf("Sending error logs to Loggly")
-		logglyClient = &logglyErrorWriter{loggly.New(logglyToken)}
-		errorOut = NonStopWriter(errorOut, logglyClient)
+		return
 	}
 
-	debugOut := timestamped(NonStopWriter(os.Stdout, file))
-	golog.SetOutputs(errorOut, debugOut)
-	return file
-}
+	if version == "" {
+		log.Error("No version configured, Loggly won't include version information")
+		return
+	}
 
-func Configure(cfg *config.Config) (err error) {
-	cfgMutex.Lock()
-	defer cfgMutex.Unlock()
+	if buildDate == "" {
+		log.Error("No build date configured, Loggly won't include build date information")
+		return
+	}
 
-	if logglyClient == nil {
-		return fmt.Errorf("No loggly client to configure.")
+	cfgMutex.Lock()
+	if cfg.Addr == lastAddr {
+		cfgMutex.Unlock()
+		log.Debug("Logging configuration unchanged")
+		return
 	}
 
+	// Using a goroutine because we'll be using waitforserver and at this time
+	// the proxy is not yet ready.
+	go func() {
```

Comment:
I moved the goroutine into here so that callers don't need to be aware of the nuance.